src/core/Tooltip.tsx

```diff
@@ -2,41 +2,77 @@ import React, {
   ButtonHTMLAttributes,
   HTMLAttributes,
   PropsWithChildren,
-  useEffect,
+  ReactNode,
+  useLayoutEffect,
   useRef,
   useState,
 } from "react";
+import { createPortal } from "react-dom";
 import Icon from "./Icon";
+import { ColorClass, Theme } from "./styles/colors/types";
+import { determineThemeColor } from "./styles/colors/utils";
 
 type TooltipProps = {
+  triggerElement?: ReactNode;
   triggerProps?: ButtonHTMLAttributes<HTMLButtonElement>;
   tooltipProps?: HTMLAttributes<HTMLDivElement>;
+  theme?: Theme;
 } & HTMLAttributes<HTMLDivElement>;
 
 const Tooltip = ({
   children,
+  triggerElement,
   triggerProps,
   tooltipProps,
+  theme = "dark",
   ...rest
 }: PropsWithChildren<TooltipProps>) => {
   const [open, setOpen] = useState(false);
+  const [fadeOut, setFadeOut] = useState(false);
   const [position, setPosition] = useState({ x: 0, y: 0 });
   const offset = 8;
   const reference = useRef<HTMLButtonElement>(null);
   const floating = useRef<HTMLDivElement>(null);
 
-  useEffect(() => {
+  const t = (color: ColorClass) => determineThemeColor("light", theme, color);
+
+  useLayoutEffect(() => {
     if (open) {
       const floatingRect = floating.current?.getBoundingClientRect();
       const referenceRect = reference.current?.getBoundingClientRect();
+      const viewportWidth = window.innerWidth;
+      const viewportHeight = window.innerHeight;
 
       if (floatingRect && referenceRect) {
-        setPosition({
-          x:
-            Math.min(floatingRect.width / 2, floatingRect.left) -
-            referenceRect.width / 2,
-          y: Math.min(floatingRect.height, floatingRect.top) + offset,
-        });
+        let x =
```

Comment:
Sorted all this tooltip instantiation logic out, it was the bare minimum before. Now it takes into account the whole viewport and its own placement with respect to the edges, and is appended to the end of the body.

Reply:
The tooltip also now has entry and exit animations